### PR TITLE
Add test for trunk feature

### PR DESCRIPTION
**Pull request by Xenwar**
Add test for trunk feature

Requires: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1125

### CODE REVIEW

File: test/e2e/suites/e2e/e2e_test.go [hunk 1]
```diff
@@ -208,8 +209,13 @@ var _ = Describe("e2e tests", func() {
 
 			shared.Byf("Creating MachineDeployment with custom port options")
 			md3Name := clusterName + "-md-3"
+
+			tTrue := true
 			customPortOptions := &[]infrav1.PortOpts{
-				{Description: "primary"},
+				{
+					Description: "primary",
+					Trunk:       &tTrue,
```
Comment (mdbooth) [inline test/e2e/suites/e2e/e2e_test.go:217]: nit: you could use `pointer.Bool(true)` here.

File: test/e2e/suites/e2e/e2e_test.go [hunk 2]
```diff
@@ -208,8 +209,13 @@ var _ = Describe("e2e tests", func() {
...
+				},
```
Comment (mdbooth) [inline test/e2e/suites/e2e/e2e_test.go:218]: Suggestion: we could have added a second port so the first would not be trunked and the second trunked.

**Review by mdbooth — APPROVED** [review]
Looks good. Some suggestions if you feel so inclined.

File: test/e2e/suites/e2e/e2e_test.go (hunk 2 above)
Comment (Xenwar) [reply test/e2e/suites/e2e/e2e_test.go:218]: Thanks, will do that.

File: test/e2e/suites/e2e/e2e_test.go (hunk 1 above)
Comment (Xenwar) [reply test/e2e/suites/e2e/e2e_test.go:217]: Thanks, will do that.

File: test/e2e/suites/e2e/e2e_test.go (hunk 2 above)
Comment (Xenwar) [reply test/e2e/suites/e2e/e2e_test.go:220]: Done, 
Tests have passed with the newly added trunked port as well.

**Review by mdbooth — APPROVED** [review]
Nice! Lets get an lgtm from somebody else.

**Pull request merged**